autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnLoadEvaluator.java

```diff
@@ -118,74 +125,105 @@ protected void execute() {
 
         Optional<Integer> mandatoryUpScaleCount = Optional.of(configMinNodeCount)
                 .filter(mandatoryUpScale -> mandatoryUpScale < 0).map(upscale -> -1 * upscale);
         Optional<Integer> mandatoryDownScaleCount = Optional.of(configMaxNodeCount)
                 .filter(mandatoryDownscale -> mandatoryDownscale < 0).map(downscale -> -1 * downscale);
 
-        YarnScalingServiceV1Response yarnResponse = yarnMetricsClient
-                .getYarnMetricsForCluster(cluster, stackV4Response, policyHostGroup, pollingUserCrn, mandatoryDownScaleCount);
+        Optional<Integer> mandatoryStopStartUpscaleCount = isStopStartEnabled ? Optional.of(stopStartAdjustment)
+                .filter(mandatoryStopStartUpscale -> mandatoryStopStartUpscale > 0) : Optional.empty();
+        Optional<Integer> mandatoryStopStartDownscaleCount = isStopStartEnabled ? Optional.of(stopStartAdjustment)
+                .filter(mandatoryStopStartDownscale -> mandatoryStopStartDownscale < 0).map(Math::abs) : Optional.empty();
+
+        YarnScalingServiceV1Response yarnResponse =
+                yarnMetricsClient.getYarnMetricsForCluster(cluster, stackV4Response, policyHostGroup, pollingUserCrn, mandatoryDownScaleCount);
 
         if (cluster.getUpdateFailedDetails() != null && pollingUserCrn.equals(cluster.getMachineUserCrn())) {
             // Successful YARN API call
             clusterService.setUpdateFailedDetails(cluster.getId(), null);
         }
 
-        int yarnRecommendedScaleUpCount = yarnResponseUtils.getYarnRecommendedScaleUpCount(yarnResponse, policyHostGroup,
-                maxAllowedUpScale, mandatoryUpScaleCount, loadAlertConfiguration.getMaxScaleUpStepSize());
-        List<String> yarnRecommendedDecommissionHosts = yarnResponseUtils.
-                getYarnRecommendedDecommissionHostsForHostGroup(cluster.getStackCrn(), yarnResponse,
-                        hostFqdnsToInstanceId, maxAllowedDownScale, mandatoryDownScaleCount, loadAlertConfiguration.getMaxScaleDownStepSize());
+        int yarnRecommendedScaleUpCount = yarnResponseUtils.getYarnRecommendedScaleUpCount(yarnResponse, policyHostGroup, maxAllowedUpScale,
+                mandatoryUpScaleCount, loadAlertConfiguration.getMaxScaleUpStepSize(), mandatoryStopStartUpscaleCount);
+        List<String> yarnRecommendedDecommissionHosts =
+                yarnResponseUtils.getYarnRecommendedDecommissionHostsForHostGroup(yarnResponse, hostFqdnsToInstanceId, stoppedHostInstanceIds,
+                        maxAllowedDownScale, mandatoryDownScaleCount, mandatoryStopStartDownscaleCount, loadAlertConfiguration.getMaxScaleDownStepSize());
 
-        LOGGER.info("yarnRecommendedScaleUpCount={}, yarnRecommendedDecommssion={}", yarnRecommendedScaleUpCount, yarnRecommendedDecommissionHosts);
+        LOGGER.info("yarnRecommendedScaleUpCount={}, yarnRecommendedDecommission={}", yarnRecommendedScaleUpCount, yarnRecommendedDecommissionHosts);
```

Comment:
Related to the comment in YarnResponseUtils. This is a misleading log message - given this isn't really a result of what YARN recommended directly. Instead it is potentially changed with things like madnatoryUp/DownScale, etc. All of this should be logged separately.

Reply:
After refactoring, this log line now pertains only to responses received from YARN